wp-sw-cache/wp-sw-cache-main.php

```diff
@@ -21,18 +22,36 @@ public static function init() {
     }
   }
 
+  public function update_version($name = '') {
+    if(!$name) {
+      $name = time();
+    }
+    update_option('wp_sw_cache_name', self::$cache_prefix.'-'.$name);
+  }
+
   public function write_sw() {
+
     $files = get_option('wp_sw_cache_files');
+    $file_keys = array();
     if(!$files) {
-        $files = array();
+      $files = array();
     }
     foreach($files as $index=>$file) {
-        $files[$index] = get_template_directory_uri().'/'.$file;
+      $tfile = get_template_directory().'/'.$file;
+
+      if(file_exists($tfile)) {
+        $file_keys[get_template_directory_uri().'/'.$file] = filemtime($tfile);
+      }
     }
 
+    $file_keys = array_keys($file_keys);
+    $name = md5(serialize($file_keys));
```

Comment:
I think you should make the md5 over the `filemtime()` returned values as they are what change. Aren't they? So your cache name is updated as soon as the modification time changes.